include/networkit/components/StronglyConnectedComponents.hpp

```diff
@@ -1,70 +1,143 @@
- * StronglyConnectedComponents.h
+ * StronglyConnectedComponents.hpp
  *
  *  Created on: 01.06.2014
- *      Author: Klara Reichard (klara.reichard@gmail.com), Marvin Ritter (marvin.ritter@gmail.com)
+ *      Authors: Klara Reichard <klara.reichard@gmail.com>
+ *               Marvin Ritter <marvin.ritter@gmail.com>
+ *               Obada Mahdi <omahdi@gmail.com>
+ *               Eugenio Angriman <angrimae@hu-berlin.de>
  */
 
+// networkit-format
+
 #ifndef NETWORKIT_COMPONENTS_STRONGLY_CONNECTED_COMPONENTS_HPP_
 #define NETWORKIT_COMPONENTS_STRONGLY_CONNECTED_COMPONENTS_HPP_
 
+#include <map>
+#include <vector>
+
+#include <networkit/base/Algorithm.hpp>
 #include <networkit/graph/Graph.hpp>
 #include <networkit/structures/Partition.hpp>
 
+#include <tlx/define/deprecated.hpp>
+
 namespace NetworKit {
 
 /**
  * @ingroup components
- * Determines the strongly connected components of an directed graph.
+ *
  */
-class StronglyConnectedComponents {
+class StronglyConnectedComponents final : public Algorithm {
+
 public:
-    StronglyConnectedComponents(const Graph& G, bool iterativeAlgo=true);
+    /**
+     * Determines the strongly connected components of a directed graph using Tarjan's algorithm.
+     *
+     * @param G Graph A directed graph.
+     */
+    StronglyConnectedComponents(const Graph &G);
 
     /**
-     * This method determines the connected components for the graph g
-     * (by default: iteratively).
+     * Determines the strongly connected components of a directed graph using Tarjan's algorithm.
+     *
+     * @param G Graph A directed graph.
+     * @param iterativeAlgo bool Whether to use the iterative algorithm or the recursive one.
+     *
+     * This constructor has been deprecated because the recursive algorithm has been deprecated.
+     */
+    TLX_DEPRECATED(StronglyConnectedComponents(const Graph &G, bool iterativeAlgo));
+
+    /**
+     * Runs the algorithm.
      */
-    void run();
+    void run() override;
 
     /**
      * This method determines the connected components for the graph g
      * (iterative implementation).
+     *
+     * This method is deprecated, run() already uses the iterative implementation.
      */
-    void runIteratively();
+    void TLX_DEPRECATED(runIteratively());
 
     /**
      * This method determines the connected components for the graph g
      * (recursive implementation).
+     *
+     * This method is deprecated, use run().
+     */
+    void TLX_DEPRECATED(runRecursively());
+
+    /**
+     * Return the number of connected components.
+     *
+     * @return count The number of components of the graph.
+     */
+    count numberOfComponents() const {
+        assureFinished();
+        return componentIndex;
+    }
+
+    /**
+     * Returns the component of the input node.
+     *
+     * @param[in] u The input node.
+     * @return count The index of the component of the input node.
      */
-    void runRecursively();
+    index componentOfNode(node u) const {
```

Comment:
Remark: Some getter-functions have a verb-based suffix, some have not. Is there any agreement on what should be used? Probably outside the scope of the PR, but if there is - maybe we should align function-names on refactoring?

Reply:
In general, I'd say that verbs are preferred but sometimes there is just no good verb (getComponentOfNode doesn't really add anything). Renaming stuff certainly has a high cost that we should not pay here (we'd need to deprecate the old method, introduce a new one, users need to update their code etc.).

Reply:
Good points.